### ISSUE TITLE
the generated source link is empty when shallow cloning

### ISSUE THREAD

**Issue by waynebrantley** [dotnet/sourcelink#251]
If shallow cloning (via Github Api) is done on a git repo, none of the source links are generated.

https://www.appveyor.com/blog/2014/06/04/shallow-clone-for-git-repositories/

**Comment by tmat** [comment]
Could you zip up and share the directory containing the shallow cloned repository?

**Comment by waynebrantley** [comment]
Unfortunately I do not - using Appveyor and they use the GitHub API to get it - but I never see it.
Turning off shallow cloning makes it work perfectly...

**Comment by tmat** [comment]
Could you perhaps add a PowerShell script to the build pipeline that zips up the source dir? 
Or perhaps AppVeyor logs git commands it executes to replicate the repo layout?

**Comment by bording** [comment]
It should be easy enough to repro this by doing your own shallow clone of a repo using `git clone --depth 1`, or whatever depth value you want to set.

**Comment by tmat** [comment]
Yeah, but I'd like to double check that that's exactly what AppVeyor does.

**Comment by bording** [comment]
Ah, yes looking at it a bit, AppVeyor has two settings. You can set a clone depth, which would use the value I mentioned above, and was the one I've interacted with before when using AppVeyor. They also support a "shallow clone" feature, which is what @waynebrantley mentions.

That is just grabbing a zip of the contents of a given commit, so it totally makes sense that Source Link wouldn't work. There is no actual git repo at that point, they are just building the code extracted from the zip file.

**Comment by bording** [comment]
The fact that it won't have a .git folder is mentioned on the first response [here].(https://help.appveyor.com/discussions/suggestions/218-from-git-perform-shallow-clone/page/1)

**Comment by tmat** [comment]
@bording Yes, not having `.git` dir would explain. Similar situation occurs when building in containers. I think for these cases we need to finish work on [Standard CI](https://github.com/dotnet/designs/blob/master/accepted/build/standard-ci-env-variables.md) variables so that the information can flow from the environment that AppVeyor sets up.

**Comment by tmat** [comment]
Another way how to make this work is to run a script that writes minimal git repository data files that Source Link requires: 
https://github.com/dotnet/sourcelink/blob/master/docs/README.md#minimal-git-repository-metadata
